### PR TITLE
Support for Android NDK r22-beta1

### PR DESCRIPTION
**Pull request by iguessthislldo**
I think builds using standalone toolchain are fine so I left them alone, but `Configure` will fail if using the NDK directly because the `platforms` and `sysroot` directories were removed.

If `sysroot` is missing, omit the `--sysroot` and `-gcc-toolchain` arguments and use the triplet form clang command.

Also since `platforms` was being used for the default API level, use `meta/platforms.json` instead if needed.

I tested this by building with r21d and r22-beta1, both using the NDK directly and using a standalone toolchain. These were just builds. I included "CLA: trivial" in my commit message, but am unsure if this is actually considered trivial or not.

### CODE REVIEW

**Comment by mattcaswell** [comment]
This looks ok to my eye, but I'd like @levitte to take a look.

I've added the "CLA: trivial" label for now, because you've flagged it as such. However I think this change goes beyond our definition of trivial and that a CLA is required:

https://www.openssl.org/policies/cla.html

**Comment by iguessthislldo** [comment]
I will submit one then. Do I need to force push to get rid of the "CLA: trivial" in the commit message afterwards or is it fine as is?

**Comment by iguessthislldo** [comment]
I went ahead and force pushed the same commit with the CLA part removed made the email address match the one I registered in the CLA.

**Review by mattcaswell — APPROVED** [review]

**Comment by mattcaswell** [comment]
Needs second review

**Review by paulidale — APPROVED** [review]

File: Configurations/15-android.conf
Comment (levitte) [inline Configurations/15-android.conf:80]: Mind changing `if ($#platforms < 0)` to `unless (@platforms)`?  The latter is much more prevalent in modern Perl (in so far that Perl 5 is modern :wink:).

**Review by levitte — APPROVED** [review]

**Comment by levitte** [comment]
I can't claim understanding the whole Android structure, but this was clear enough.  Good work!

**Comment by levitte** [comment]
Merged

948fd7af627f42cf9ab6b7c3df34aa5ebad422ae Support for Android NDK r22-beta1

**Comment by wongsyrone** [comment]
Please backport this to OpenSSL 1.1.1 branch.
#13685

**Comment by wongsyrone** [comment]
I fixed it on my side using #13694. Works smoothly. Good job.

**Pull request closed without merging**
